### DISCUSSION TITLE
Notification to 'emitSelf'?

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#5714] [Help]
I can see that we can emit an event for a notification action:

```
Notification::make()
    ->title('Are You Sure?')
    ->body('Remove the item?')
    ->danger()
    ->actions([
        Action::make('yes')
            ->label('Yes, remove it')
            ->emit('removeCartItem'),
        Action::make('no')
            ->label('Keep it in the cart')
            ->close()
    ])
    ->send();
```
However I wondered if it was possible to `emitSelf`. This example is part of a `CartItem` livewire component. Emitting the `removeCartItem` event triggers it on all loaded insrances of `CartItem` which results in all items being removed from the cart.

There are ways around this:

1. Dynamically generate an event name for each cart item
2. Pass all relevant data via the event (but this will still result in the event firing for each cart item)

Is it currently possible to trigger "emitting to self"? Would a PR be welcome?

**Comment by danharrin** [comment]
Maybe an `isSelf` parameter would be good

**Reply by titantwentyone** [reply]
I've used `shouldEmitToSelf` purely because it just makes it simpler to read and have a PR ready to go but wondering if you want me to expand this so that it can handle `emitTo` and `emitUp` as well as `emitSelf`? Thinking the parameter could be renamed to `emitDirection`?

**Reply by danharrin** [reply]
I think it is fine to have separate methods for To and Up and Self

**Reply by titantwentyone** [reply]
PR opened #5728
